eth/vm/state.py

```diff
@@ -51,7 +51,7 @@ def __init__(
             self,
             db: AtomicDatabaseAPI,
             execution_context: ExecutionContextAPI,
-            state_root: bytes) -> None:
+            state_root: Hash32) -> None:
```

Comment:
@carver This is the example of cargo-culting I mentioned on gitter. `tox` said (before the change):

```
eth/vm/state.py:57: error: Argument 2 to "AccountDatabaseAPI" has incompatible type "bytes"; expected "Hash32"
```

... so I gave it exactly what it asked. But I don't know if it's right or not.

Reply:
Yeah, in this case, it looks like the right thing to do to go with `Hash32`, since it's logically the hash of the root node. :+1:

Reply:
Should `has_root()` of `AccountDB` (and ABC `AccountDatabaseAPI`) have a `state_root` argument that's `Hash32` also?..

https://github.com/ethereum/py-evm/blob/0d79bb6c56f4835ef9242cbad18a51ef86e77011/eth/db/account.py#L135-L136